rts/lra/client/src/main/java/io/narayana/lra/client/NarayanaLRAClient.java

```diff
@@ -734,18 +747,23 @@ private URI enlistCompensator(URI uri, Long timelimit, String linkHeader, String
 
             String responseEntity = response.hasEntity() ? response.readEntity(String.class) : "";
             if (response.getStatus() == Response.Status.PRECONDITION_FAILED.getStatusCode()) {
-                LRALogger.i18nLogger.error_tooLateToJoin(lraId, responseEntity);
-                String errorMsg = lraId + ": Too late to join with this LRA";
+                String logMsg = LRALogger.i18nLogger.error_tooLateToJoin(lraId, responseEntity);
+                LRALogger.logger.error(logMsg);
+                String errorMsg = String.format("%s", logMsg);
                 throw new WebApplicationException(errorMsg,
                         Response.status(PRECONDITION_FAILED).entity(errorMsg).build());
             } else if (response.getStatus() == NOT_FOUND.getStatusCode()) {
-                LRALogger.i18nLogger.info_failedToEnlistingLRANotFound(
+                String logMsg = LRALogger.i18nLogger.info_failedToEnlistingLRANotFound(
                         lraId, coordinatorUrl, NOT_FOUND.getStatusCode(), NOT_FOUND.getReasonPhrase(), GONE.getStatusCode(), GONE.getReasonPhrase());
-                throw new WebApplicationException(uri.toASCIIString(),
+                LRALogger.logger.info(logMsg);
+                String errorMsg = String.format("%s (%s)", logMsg, uri.toASCIIString());
+                throw new WebApplicationException(errorMsg,
```

Comment:
There is a mismatch here, in other places the exception message and the response to the client are the same text. For consistency I would keep them the same. Similarly in other places in this PR.

Reply:
Note that ,although you appear to be adding extra information (uri.toASCIIString()) to the exception message, the uri is actually the same as the lraId which has already been added to logMsg, ie it adds no value. Therefore using the value in the variable logMsg is sufficient.

And I don't recall (without checking) what I was referring to when I said `Similarly in other places in this PR` so you should perform a quick sanity check to verify you haven't done something similar elsewhere.